### PR TITLE
Add management of jisc mail subscriptions

### LINKED ISSUE #748: Keep JISCMail in sync
We need to propagate subscriptions/unsubscriptions to JISCMail by emailing their listserv.

### PR DESCRIPTION
**Pull request by edpaget**
Closes #748

### CODE REVIEW

**Comment by edpaget** [comment]
I think [this](https://github.com/edpaget/Panoptes/blob/jisc-mail-workers/app/controllers/api/v1/users_controller.rb#L31) may be some bug in active record. I'll try to track it down after the 21st.

File: app/workers/subscribe_worker.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class SubscribeWorker
+  include Sidekiq::Worker
+
+  def perform(email, display_name)
+    JiscMailer.subscribe(email, display_name).deliver
```
Comment (camallen) [inline app/workers/subscribe_worker.rb]: Do we want to check which env we are in? I.e only run these in prod.

**Comment by camallen** [comment]
looks good.

@astopy does Jisc mail require a certain email address / domain to sub / unsub these accounts? Pretty sure these will come from no-reply@zooniverse.org.

**Comment by astopy** [comment]
no-reply@zooniverse.org <mailto:no-reply@zooniverse.org> will be fine I think. I’ll just need to add the right password to the production config.

File: app/workers/subscribe_worker.rb (hunk 1 above)
Comment (edpaget) [inline app/workers/subscribe_worker.rb]: Sure I'll make that change.

File: app/workers/unsubscribe_worker.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class UnsubscribeWorker
+  include Sidekiq::Worker
+
+  def perform(email)
+    JiscMailer.unsubscribe(email).deliver
```
Comment (camallen) [inline app/workers/unsubscribe_worker.rb]: And here too?

**Comment by astopy** [comment]
The password/config should be set up correctly now, ready for when this is live.

**Pull request merged**